vpdq/cpp/regtest.py

```diff
@@ -72,91 +74,114 @@ def get_argparse() -> argparse.ArgumentParser:
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-
-    cdir = os.getcwd()
-    pdir = os.path.dirname(cdir)
-    sample = pdir + ("/sample-hashes")
-    output = outputHashFolder
     distanceTolerance = str(args.matchDistanceTolerance)
     qualityTolerance = str(args.qualityTolerance)
-    for file in os.listdir(sample):
-        if file.endswith(".txt"):
-            print("\nMatching File " + file)
-            sampleFile = f"{sample}/{file}"
-            outputFile = f"{output}/{file}"
+    verbose = args.verbose
+
+    if not inputVideoFolder.exists():
+        raise argparse.ArgumentTypeError(
+            f"inputVideoFolder: {inputVideoFolder} does not exist."
+        )
+
+    with TemporaryDirectory() as tempOutputHashFolder:
+        """
+        Write the files to temp
+        If outputHashFolder is specified then
+        copy the output files there, overwriting existing files
+        """
+
+        tempOutputHashFolder = Path(tempOutputHashFolder)
+        if outputHashFolder is not None:
+            if not outputHashFolder.exists():
+                print(f"Creating output directory at {outputHashFolder}")
+                outputHashFolder.mkdir(parents=True, exist_ok=True)
+            print(f"Writing output to directory: {outputHashFolder}")
+        else:
+            print(f"Writing output to temp directory: {tempOutputHashFolder}")
+        # TODO: Add more general options for other video extensions.
+        for fileStr in glob.iglob(f"{inputVideoFolder}/**/*.mp4", recursive=True):
+            file = Path(fileStr)
+
+            """
+            Create output hash file or overwrite existing file
+            This is hardcoded in cpp, and it
+            does not create the file if it does not exist:
+
+            vpdq-hash-video.cpp:
+
+            // Strip containing directory:
+            std::string b = basename(inputVideoFileName, "/");
+            // Strip file extension:
+            b = stripExtension(b, ".");
+            outputHashFileName = outputDirectory + "/" + b + ".txt";
+            """
+            outputHashFile = tempOutputHashFolder / f"{file.stem}.txt"
+            with open(outputHashFile, "x+t"):
+                pass
```

Comment:
This might be surprising enough to a reader that something like

```
pass # opening a file in write mode truncates it
```

might be helpful for code historians of the future

Reply:
No, it does not truncate it. `x` does not truncate, `w` does. Additionally, this is for the temporary directory, so if there is somehow a name collision there are bigger problems than truncating the file.

If an output directory is specified, _copying_ the file will replace it if it exists in the outputHashFolder, which I cannot imagine anyone would expect otherwise. If it was a regular CLI program it would probably be helpful to add some warning messages, but this is a simple regression test script, and it would be very irritating to have to delete the output folder contents every time the test needs to be run.

Reply:
It could be just x because it's not read from. It could also be w.

I just used t to specify it's a text file. Doesn't affect anything since nothing is written to it.

Writing to temp then copying does mean on a very early failure it will be cleaned up.

Yes you could use touch. I didn't know that was a Pathlib method. I'll change it.